### PR TITLE
Feature/opentelemetry tracing

### PR DESCRIPTION
**Pull request by Sachin4403**

### CODE REVIEW

File: candlelight.go
Comment (schmidtw) [inline candlelight.go:44]: Should the configuration return an error or nil to indicate if there was an issue?

Comment (Sachin4403) [reply candlelight.go:44]: we can't return nil if the configuration is missing in properties.  if we are doing this then we will have issues when we are starting span. I had written the code in such a way if nothing is specified it will be taking the default as stdout and will be creating the global tracer based on that.

example usage: candlelight.ConfigureTracerProvider(v.Sub("traceProvider"),logger,applicationName)
if traceProvider is missing in the properties file then viper will be coming as nil in this case we are creating the stdout tracer.

Yes, we can return an error that will not start the application.


What do you suggest should we continue like this or we should return an error which will not start the application?

Comment (schmidtw) [reply candlelight.go:44]: Of the two I'd return an error and not start the application to help prevent a silent failure that could be more costly (detected when you need it at runtime).

**Review by kristinaspring — APPROVED** [review]
Tracing is a valuable addition, and we're excited to see it in our services! The team discussed and we agreed this is a great starting point for a feature branch. Once this is merged to the branch, we will create issues and then work with you to fix them before merging the feature branch into the main branch.

Unfortunately, we can't merge this until the CLA Assistant passes.  It looks like the problem is the author in the commits hasn't signed the CLA - I think that's because the author needs to be updated.  This stackoverflow answer explains how you can do that:
https://stackoverflow.com/a/3042512

Once that's done, I'll happily merge your pr. 🙂

File: candlelight.go
Comment (Sachin4403) [reply candlelight.go]: I had updated the code as per this conversation. now we need to handle the error where we are using this module.

**Pull request merged**
